File: Algorithms/169.MajorityElement.py
__author__ = 'qianlong'
class Solution:
    # @param nums, a list of integers
    # @return an integer
    def majorityElement(self, nums):
        previous = None
        count = 0
        for current in nums:
            if count == 0:
                previous = current
                count += 1
            else:
                if current == previous:
                    count += 1
                else:
                    count -= 1
        return previous

File: Algorithms/169.MajorityElement.java
public class Solution {
    public int majorityElement(int[] nums) {
        int previous = 0;
        int count = 0;
        for (int i = 0; i < nums.length; i++) {
            if (count == 0) {
                previous = nums[i];
                count ++;
            } else {
                if (previous == nums[i]) {
                    count ++;
                } else {
                    count --;
                }
            }
        }
        return previous;  
    }
}
